src/v/cluster/self_test_rpc_types.cc

```diff
@@ -35,6 +35,26 @@ std::ostream& operator<<(std::ostream& o, self_test_status sts) {
     return o;
 }
 
+ss::sstring self_test_stage_as_string(self_test_stage sts) {
+    switch (sts) {
+    case self_test_stage::idle:
+        return "idle";
+    case self_test_stage::disk:
+        return "disk";
+    case self_test_stage::net:
+        return "net";
+    case self_test_stage::cloud:
+        return "cloud";
+    default:
+        __builtin_unreachable();
```

Comment:
you can probably drop builtin_unreachable if the cases are a covering set for the enum. if they aren't a covering set and you want to fail on default, then use vassert. if they are a covering set but you get a compiler warning, move builtin_unreachable outside the switch statement.